Automattic/newspack-plugin
# Campaigns Wizard: favorite category segmentation

### All Submissions:






### Changes proposed in this Pull Request:



UI for adding  a favourite-category based segmentation. 

Do not merge until https://github.com/Automattic/newspack-popups/pull/322 is merged.

### How to test the changes in this Pull Request:

1. Switch Campaign plugin to `add/handle-category-segment` branch (https://github.com/Automattic/newspack-popups/pull/322)
1. Visit the Campaigns Wizard, Segmentation tab. Create or edit a segment
2. Observe a new section: "Category Affinity". It should list all categories, and if there are more than 5, allow for list showing/hiding the whole list with a button
3. Assign a favourite category to the segment, and create a campaign targeted to that segment
4. Verify that the campaign is shown only if the reader did most of their reading in the selected category
5. Now create a segment with multiple favourite categories
6. Verify that the campaign is displayed only if the reader did most of their reading in one of the selected categories
7. Create a segment combining favourite categories with other conditions, like minimum amount of posts read
8. Verify the campaign is shown only of all segment conditions are met
9. Create a segment without favourite categories, verify it works as expected (just to make sure an empty favourite categories set will not break things)

### Other information:
